src/StreetNameRegistry.Api.BackOffice/Infrastructure/Authorization/StreetNameNisCodeFinder.cs

```diff
@@ -0,0 +1,25 @@
+namespace StreetNameRegistry.Api.BackOffice.Infrastructure.Authorization
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.EntityFrameworkCore;
+    using StreetNameRegistry.Api.BackOffice.Abstractions;
+    using StreetNameRegistry.Municipality;
+
+    public class StreetNameNisCodeFinder : INisCodeFinder<PersistentLocalId>
+    {
+        private readonly BackOfficeContext _backOfficeContext;
+
+        public StreetNameNisCodeFinder(BackOfficeContext backOfficeContext)
+        {
+            _backOfficeContext = backOfficeContext;
+        }
+
+        public async Task<string?> FindAsync(PersistentLocalId streetNamePersistentLocalId,  CancellationToken ct)
+        {
+            var result = await _backOfficeContext.FindMunicipalityIdByPersistentLocalId(streetNamePersistentLocalId, ct);;
+
+            return result?.NisCode;
```

Comment:
Let's monitor this in staging to see what the performance hit is.
Possibly need a caching mechanism.

Reply:
It's a DB call we were already performing, but than further down the process: https://github.com/Informatievlaanderen/streetname-registry/blob/main/src/StreetNameRegistry.Api.BackOffice/Handlers/ApproveStreetNameHandler.cs#L28

Because we twice use DbSet.Find(), we'll be using the EF cache.